client/src/app/site/agenda/services/agenda-pdf.service.ts

```diff
@@ -0,0 +1,113 @@
+ * ```
+ */
+@Injectable({
+    providedIn: 'root'
+})
+export class AgendaPdfService {
+    /**
+     * Constructor
+     *
+     * @param translate handle translations
+     * @param treeService create hierarchy between items
+     * @param ConfigService for fetching the event name
+     */
+    public constructor(
+        private translate: TranslateService,
+        private treeService: TreeService,
+        private config: ConfigService
+    ) {}
+
+    /**
+     * Creates pdfMake definitions for a agenda list pdf from the given agenda items
+     *
+     * @param items A list of viewItems to be included in this agenda list. Items with the property 'hidden'
+     * will be ignored, all other items will be sorted by their parents and weight
+     * @returns definitions ready to be opened or exported via {@link PdfDocumentService}
+     */
+    public agendaListToDocDef(items: ViewItem[]): object {
+        const tree: OSTreeNode<ViewItem>[] = this.treeService.makeTree(items, 'weight', 'parent_id');
+        const title = {
+            text: this.translate.instant('Agenda'),
+            style: 'title'
+        };
+        const subtitle = {
+            text: this.config.instant<string>('general_event_name'),
+            style: 'subtitle'
+        };
+        const entries = this.createEntries(tree);
+        return [title, subtitle, entries];
+    }
+
+    /**
+     * Traverses the given nodeTree and creates an array of entries for all items
+     *
+     * @param tree
+     * @returns hierarchical pdfMake definitions for topic entries
+     */
+    private createEntries(tree: OSTreeNode<ViewItem>[]): AgendaTreePdfEntry[] {
+        const content: AgendaTreePdfEntry[] = [];
+        tree.forEach(treeitem => content.push(...this.parseItem(treeitem, 0)));
+        return content;
+    }
+
+    /**
+     * Parses an entry line and triggers parsing of any children
+     * (TODO: Check assumption: items with 'is_hidden' are not to be exported)
+     *
+     * @param nodeItem the item for the head line
+     * @param level: The hierarchy index (beginning at 0 for top level agenda topics)
+     * @returns pdfMake definitions for the number/title strings, indented according to hierarchy
+     */
+    private parseItem(nodeItem: OSTreeNode<ViewItem>, level: number): AgendaTreePdfEntry[] {
+        const itemList: AgendaTreePdfEntry[] = [];
+        if (!nodeItem.item.item.is_hidden) {
+            // don't include hidden items and their subitems
+            const resultString: AgendaTreePdfEntry = {
+                style: level ? 'listChild' : 'listParent',
+                columns: [
+                    {
+                        width: level * 15,
+                        text: ''
+                    },
+                    {
+                        width: 60,
+                        text: nodeItem.item.itemNumber
+                    },
+                    {
+                        text: nodeItem.item.getListTitle()
```

Comment:
getListTitle contains also the itemNumber which is already in your first column!

Reply:
Please use a bigger font size for items - like in 2.3 agenda pdf.